File: src/workflow/src/url.js
/*
 *
 * Copyright © 2019 Yokesh Thirumoorthi.
 *
 * [This program is licensed under the "MIT License"]
 * Please see the file LICENSE in the source
 * distribution of this software for license terms.
 *
 */

const {CADDT_SERVER_ENDPOINT} = require('./common/config');
const signature = require('./signature');
const {
  RESIZE_CORE_ENDPOINT,
  CADDT_SERVER_ENDPOINT
} = require('./common/config');

const UPLOADS_VOL = 'album-uploads';
const RESIZED_VOL = 'album-resized';
const FACES_VOL = 'album-faces';

const uploadURL = (albumName, photoName) => `http://${CADDT_SERVER_ENDPOINT}/${UPLOADS_VOL}/${albumName}/uploads/${photoName}`;
const resizedURL = (albumName, photoName) => `http://${CADDT_SERVER_ENDPOINT}/${RESIZED_VOL}/${albumName}/${photoName}`;
const facesURL = (albumName, photoName) => `http://${CADDT_SERVER_ENDPOINT}/${FACES_VOL}/${albumName}/${photoName}`;

/**
 * Generate a signed imgproxy URL for resizing images.
 * The source image URL is a caddy URL.
 */
function getImgProxyPhotoResizeURL(caddyURL) {
    const signedUrlSegment = signature.getSignedImgURL(caddyURL);
    const URL = `http://${RESIZE_CORE_ENDPOINT}${signedUrlSegment}`;
    return URL;
}

module.exports = {
    uploadURL,
    resizedURL,
    facesURL,
    getImgProxyPhotoResizeURL
}

File: src/workflow-go/utils/url.go
package utils

import (
	"fmt"
	pb "github.com/kuruvi-bits/transform/pb"
)

const (
	CADDT_SERVER_ENDPOINT = "192.168.1.100:2015"
	RESIZE_SERVER_ENDPOINT = "192.168.1.100:8080"
	UPLOADS_VOL = "album-uploads"
	RESIZED_VOL = "album-resized"	
	FACES_VOL = "album-faces"
)

type Message struct {
	AlbumName string `json:"albumName"`
	PhotoName string `json:"photoName"`
}

type URL struct {
	Upload string
	Resized string
	Faces string
}

func GetURL(message Message) URL {
	albumName := message.AlbumName
	photoName := message.PhotoName
	server := "http://" + CADDT_SERVER_ENDPOINT + "/" 
	uploadURL := server + UPLOADS_VOL + "/" + albumName + "/uploads/" + photoName
	resizedURL := server + RESIZED_VOL + "/" + albumName + "/" + photoName
	facesURL := server + FACES_VOL + "/" + albumName + "/" + photoName
	fmt.Println(uploadURL)
	url := URL{
		Upload: uploadURL,
		Resized: resizedURL,
		Faces: facesURL,
	}
	return url
}

func GetResizeURL(message Message) string {
	caddyURL := GetURL(message).Upload
    fmt.Println("Caddy URL", caddyURL)
	signedURL := GetSignedPathForResize(caddyURL)
	resizedURL := "http://" + CADDT_SERVER_ENDPOINT + "/" + signedURL
	return resizedURL
}

func GetFaceCropURL(message Message, boundingBox *pb.BoundingBox) string {
	caddyURL := GetURL(message).Resized
    fmt.Println("Caddy URL", caddyURL)
	signedURL := GetSignedPathForFaceCrop(caddyURL, boundingBox)
	resizedURL := "http://" + CADDT_SERVER_ENDPOINT + "/" + signedURL
	return resizedURL
}

func GetResizedVolPath(message Message) string {
	path := RESIZED_VOL + "/" + message.AlbumName + "/" + message.PhotoName
	return path
}